### DISCUSSION TITLE
Integration hacs-ecoflow cloud gets unavailable  or partial data

### DISCUSSION THREAD

**Discussion by PhilNS** [tolwi/hassio-ecoflow-cloud#302] [General]
Since a few weeks the integration keeps failing to retrieve some or all data from Ecoflow continuously.
 I get Unavailable data , especially from all sensors, while for example the switches are (randomly) working, i.e. AC on and off.

HA reports this before and after I manually trigger an ecoflow (AC) switch:

1) Status changed to assume_offline       - 09:43:52 
2) AC Enabled turned off Switch: Turn off - 09:59:05  - Phil
3) Status changed to assume_online        - 09:59:05  
4) AC Enabled turned on Switch: Turn on   - 09:59:13  - Phil
5) Status changed to assume_offline       - 09:59:22 
     -- and then goes back as status assumed offline even if AC and device are on.
6)  Status changed to assume_offline      - 10:24:56 

The integration DEBUG at this stage is like the attached : 
0 - AFTER SWITCH TRIGGER: ---------------------------------------
[config_entry-ecoflow_cloud-01J6BX1P4744NTKJ7S4MR5M69A - 0 after switch trigger.json](https://github.com/user-attachments/files/16780995/config_entry-ecoflow_cloud-01J6BX1P4744NTKJ7S4MR5M69A.-.0.after.switch.trigger.json)

Then if I remove and reinstall the integration, I initially get no data /unavailable data
The integration DEBUG at this stage is like the attached below  :

1 - START NO VALUES -----------------------------------------------

[config_entry-ecoflow_cloud-01J6BX1P4744NTKJ7S4MR5M69A - 1 start-novalues.json](https://github.com/user-attachments/files/16780994/config_entry-ecoflow_cloud-01J6BX1P4744NTKJ7S4MR5M69A.-.1.start-novalues.json)

And If I then open the official Ecoflow app on android device, then I immediately get the full data into the integration which though last for some time and then they get stuck again

Here again HA logs

6) Status changed to assume_online        - 10:15:17 
7) X-Boost Enabled turned on                  - 10:15:24 
8) AC Enabled turned on                           - 10:15:24 
9) AC Always On turned on                      - 10:15:24 
...
10)  Status changed to assume_offline      - 10:24:56 (even if AC and Ecoflow device is on )

The integration DEBUG at this stage is like the attached below  : 

2 - AFTER ANDROID APP OPEN DEVICE ----------------------------

[config_entry-ecoflow_cloud-01J6BX1P4744NTKJ7S4MR5M69A - 2 after android app open.json](https://github.com/user-attachments/files/16780993/config_entry-ecoflow_cloud-01J6BX1P4744NTKJ7S4MR5M69A.-.2.after.android.app.open.json)

I really can't phatom what is impacting on the continuous stable flow of correct data.

Any idea on what could cause this ?

thanks
Phil

**Comment by denbe52** [comment]
I have exactly the same problem - running Home Assistant on a Raspberry Pi.
Just tried installing latest ecoflow-cloud version 1.1.0 that was released on Sept 5 - maybe that will help.....

**Comment by Nid01** [comment]
@PhilNS @denbe52 
EcoFlow modified the app API so it only provides data as long as the app is opened.

Here is the info provided via Facebook:
https://www.facebook.com/groups/1405868123482532/posts/1714184662650875/
> For those who haven't obtained an official AccessKey and SecretKey:
> If your integration relies on code from sources that do not require official credentials, such as GitHub, please be aware of the significant risks involved. As our user base continues to grow, the additional strain on our servers may necessitate further measures against unauthorized clients. We recognize that this could cause inconvenience and apologize in advance for any disruptions this may cause.
> Starting in early August, connections from unauthorized clients have been restricted. Specifically, MQTT clients that mimic app behavior are no longer recognized as valid app clients. As a result, data updates will not occur when the EcoFlow app is not actively running.
> We strongly encourage you to register as an official developer and apply for an AccessKey and SecretKey. The EcoFlow API currently supports devices such as the DELTA 2, DELTA 2 Max, DELTA Pro, SHP, PowerOcean, and PowerStream. Please refer to our documentation for the most up-to-date list of supported devices, as we continue to expand our offerings.

The integration is currently being updated to use the public API.

**Reply by denbe52** [reply]
Thank you for the update - it's been working fine so far with ecoflow-cloud version 1.1.0.
However, I have applied for the API keys and will update my HA once I get them.

**Reply by AlphaZuluNRW** [reply]
Does anyone have a solution on how to get rid of the "unknown" error message when using the API keys? I've already tried generating new keys, but that leads to the same result in HA.

**Comment by PhilNS** [comment]
Thanks . Indeed version 1.1.x has solved the problem, now keeping the data updates alive. Let's see if that EcoFlow company policy modification will impact also this version requiring a dedicated personal developer API (to be placed somewhere then).
